proposals/expand-ref.md

```diff
@@ -0,0 +1,294 @@
+    Span<int> local3 = p1; 
+    Span<int> local4 = p2; 
+
+    // Okay: the safe-to-escape of the value is inferred in this case as it is 
+    // done for ref locals today.
+    ref Span<int> refLocal1 = ref p1;
+    ref Span<int> refLocal2 = ref p2;
+}
+```
+
+A `ref scoped` parameter is also implicitly `scoped ref`. That means neither the value nor it's `ref` can be returned from the method. Both `ref` and `in` parameters can have their values modified with `scoped`. An `out` parameter cannot have its value modified with `scoped` as such a declaration is non-sensical. 
+
+```csharp
+void M(
+    ref scoped Span<int> p1,    // Okay
+    in scoped Span<int> p2,     // Okay
+    out scoped Span<int> p2,    // Error
+)
+```
+
+The [method arguments must match](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/low-level-struct-improvements.md#rules-method-arguments-must-match) rules will be updated to take `ref scoped` into account. Values passed to such parameters do not need to be considered when calculating the return scopes.
+
+Detailed notes:
+- A `ref scoped` parameter is implicitly `scoped ref`
+- A `out scoped` parameter declaration is an error
+
+### ref field to ref struct
+The language will allow for `ref struct` to appear as `ref scoped` fields. This `scoped` will serve to ensure the values cannot be escaped outside the containing instance but can be read and manipulated within it. 
+
+```csharp
+ref struct Deserializer
+{
+    ref scoped Utf8JsonReader reader;
+
+    ReadOnlySpan<byte> M1()
+    {
+        // okay: implicitly scoped to current method
+        var span = reader.ValueSpan; 
+
+        // okay
+        reader.Skip();
+
+        // Error: can't escape the ref data the ref scoped field refers to
+        return reader.ValueSpan;
+    }
+}
+```
+
+This is accomplished by giving every `ref scoped` field a new two new escape scopes named _current field N_ and _current ref field N_ where _N_ is the numeric order of the field. For example the first field has a _safe-to-escape_ of _current field 1_ and a _ref-safe-to-escape_ of _current ref field N_. Both escape scopes can be converted to _current method_, and _current field N_ can be converted to _current ref field N_, but no other defined relationships exist. That serves to restrict their usage to the current method where the containing value is used. This escape scope applies to both 
+
+Below are a few examples of these rules in action
+
+```csharp
+ref struct NestedRefStruct { }
+ref struct RefStruct
+{
+    public NestedRefStruct NestedField;
+}
+
+ref struct S
+{
+    ref scoped RefStruct field;
+
+    RefStruct M1(RefStruct s)
+    {
+        // Okay
+        field = new(); 
+
+        // Error: calling-method is not convertible to current-field-1 as they have 
+        // no relationship
+        field = s;
+
+        // Error: safe-to-escape is current-field-1 which isn't returnable 
+        return field;
```

Comment:
Actually I think these returns can be allowed. I need to tweak the rules slightly so that `current-param-N` is related to `this` not `local`.

Reply:
Never mind, they cannot I was wrong.